CMake minimum version

I think the CMake minimum required version should be set to 3.12 because of using target_link_libraries with object libraries. 

I got the following on CMake v 3.10 :

```
CMake Error at CMakeLists.txt:58 (target_link_libraries):
  Object library target "engine" may not link to anything.
```
Which is gone by updating to the latest version. 

More on this:
https://gitlab.kitware.com/cmake/cmake/-/issues/18010
https://gitlab.kitware.com/cmake/cmake/-/issues/14778

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@
 #
 # For many purposes, you may not need to change anything about this file.
 
-cmake_minimum_required(VERSION 3.8.2)
+cmake_minimum_required(VERSION 3.12)
 
 # Set project name, version and laguages here. (change as needed)
 # Version numbers are available by including "exampleConfig.h" in 

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.8.2)
+cmake_minimum_required(VERSION 3.12)
 
 # List all files containing tests. (Change as needed)
 set(TESTFILES        # All .cpp files in tests/

